Jenkinsfile

```diff
@@ -0,0 +1,246 @@
+echo "Start to build"
+
+
+properties ([
+    buildDiscarder(
+        logRotator(
+            artifactDaysToKeepStr: '',
+            artifactNumToKeepStr: '',
+            daysToKeepStr: '30',
+            numToKeepStr: '100'
+        )
+    )
+])
+
+
+parallelSteps = [:]
+parallelSteps['linux'] = streamStep('linux')
+parallelSteps['windows'] = streamStep('windows')
+
+
+def streamStep(nodeType) {
+    return {
+        node("mbed-flasher-${nodeType}") {
+            deleteDir()
+
+            try {
+                baseBuild()
+            } catch(err) {
+                throw err
+            } finally {
+                // clean up
+                step([$class: 'WsCleanup'])
+            }
+        }
+    }
+}
+
+
+def baseBuild() {
+    dir('mbed-flasher') {
+        // checkout scm
+        echo "checkout scm start"
+        checkout scm
+
+        // get commit hash
+        getGitCommit()
+
+        // copy artifacts
+        copyArtifacts()
+
+        if (isUnix()) {
+            unittest("py2")
+            unittest("py3")
+            linux_pylint_check()
+        }
+        else {
+            winTest("py2")
+            winTest("py3")
+        }
+
+        postBuild()
+    }
+}
+
+
+def getGitCommit() {
+    if (env.GIT_COMMIT_HASH == null) {
+        if (isUnix()) {
+            env.GIT_COMMIT_HASH = sh(script: 'git rev-parse HEAD', returnStdout: true).trim()
+        }
+        else {
+            env.GIT_COMMIT_HASH = bat(script: 'git rev-parse HEAD', returnStdout: true).trim()
+        }
+    }
+}
+
+
+def copyArtifacts(){
+    echo "Copying artifacts from mbed-client-cliapp_for_CI/label=GCC_ARM"
+    step([
+        $class: 'CopyArtifact',
+        fingerprintArtifacts: true,
+        flatten: true,
+        projectName: "mbed-client-cliapp_for_CI/label=GCC_ARM",
+        filter: '**/mbed-client-cliapp.bin',
+        selector: [
+            $class: 'StatusBuildSelector',
+            stable: true
+        ],
+        target: 'test/'
+    ])
+}
+
+
+def setBuildStatus(String state, String context, String message) {
+    step([
+        $class: "GitHubCommitStatusSetter",
+        reposSource: [
+            $class: "ManuallyEnteredRepositorySource",
+            url: "https://github.com/ARMmbed/mbed-clitest.git"
+        ],
+        contextSource: [
+            $class: "ManuallyEnteredCommitContextSource",
+            context: context
+        ],
+        errorHandlers: [[
+            $class: "ChangingBuildStatusErrorHandler",
+            result: "UNSTABLE"
+        ]],
+        commitShaSource: [
+            $class: "ManuallyEnteredShaSource",
+            sha: env.GIT_COMMIT_HASH
+        ],
+        statusResultSource: [
+            $class: 'ConditionalStatusResultSource',
+            results: [
+                [
+                    $class: 'AnyBuildResult',
+                    message: message,
+                    state: state
+                ]
+            ]
+        ]
+     ])
```

Comment:
A very minor nitpick: This bracket level is intended by 5 spaces and  the next by 3 instead of the 4 otherwise used.

Reply:
fixed.